front/components/data_source/DataSourcePicker.tsx

```diff
@@ -24,62 +30,89 @@ export default function DataSourcePicker({
     dataSources: { workspace_id: string; data_source_id: string }[]
   ) => void;
 }) {
-  const hasDataSource =
+  const hasDataSourceView =
     currentDataSources.length > 0 &&
     currentDataSources[0].workspace_id &&
     currentDataSources[0].workspace_id.length > 0 &&
     currentDataSources[0].data_source_id &&
     currentDataSources[0].data_source_id.length > 0;
 
-  const [name, setName] = useState(
-    hasDataSource ? currentDataSources[0].data_source_id : null
-  );
-
-  const { dataSources, isDataSourcesLoading, isDataSourcesError } =
-    useDataSources(owner, { disabled: readOnly });
+  const {
+    vaultDataSourceViews,
+    isVaultDataSourceViewsLoading,
+    isVaultDataSourceViewsError,
+  } = useVaultDataSourceViews({
+    vaultId: vault.sId,
+    workspaceId: owner.sId,
+  });
 
   const [searchFilter, setSearchFilter] = useState("");
-  const [filteredDataSources, setFilteredDataSources] = useState(dataSources);
+  const [filteredDataSourceViews, setFilteredDataSourceViews] =
+    useState(vaultDataSourceViews);
+
+  const selectedDataSourceView = hasDataSourceView
```

Comment:
Probably worth a small comment here since this is what does all the conversion magic